src/utils/logger.ts

```
@@ -13,25 +88,158 @@ export class Logger {
 
-  error(message: string, meta?: Record<string, unknown>): void {
-    console.error('\x1b[31m%s\x1b[0m', 'ERROR:', message);
-    if (meta) console.error(meta);
+  /**
+   * Starts a loading indicator with progress
+   * @param message The message to display
+   * @param total The total number of items to process
+   */
+  static startLoading(message: string, total: number): void {
+    this.loadingMessage = message;
+    this.loadingTotal = total;
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)